DOC: Added docstrings to fixtures defined in XML I/O, util and pandas modules

CristianoS16 opened the pull request:
Partially addresses: #19159

Co-authored-by: @GeorgeJuniorGG and @MylenaRoberta.

CristianoS16 committed:
docs: add documentation to xml conftest file fixtures

Co-authored-by: GeorgeJuniorGG <g216741@dac.unicamp.br>
Co-authored-by: MylenaRoberta <m222687@dac.unicamp.br>

CristianoS16 committed:
docs: add documentation to util conftest file fixtures

Co-authored-by: GeorgeJuniorGG <g216741@dac.unicamp.br>
Co-authored-by: MylenaRoberta <m222687@dac.unicamp.br>

CristianoS16 committed:
docs: add documentation to pandas conftest arrow_string_storage fixture

Co-authored-by: GeorgeJuniorGG <g216741@dac.unicamp.br>
Co-authored-by: MylenaRoberta <m222687@dac.unicamp.br>

datapythonista reviewed (commented):
Thanks for the contribution @CristianoS16, good work, I added some suggestions that you can have a look at. Note that I only add the suggestions once, but for example having the constant values `True` or `False` with backticks is something that can apply to several docstring in your changes.

CristianoS16 committed:
doc: apply suggestions

Co-authored-by: GeorgeJuniorGG <g216741@dac.unicamp.br>
Co-authored-by: MylenaRoberta <m222687@dac.unicamp.br>

CristianoS16 commented:
> Thanks for the contribution @CristianoS16, good work, I added some suggestions that you can have a look at. Note that I only add the suggestions once, but for example having the constant values `True` or `False` with backticks is something that can apply to several docstring in your changes.

Thanks @datapythonista for the suggestions. I think we fixed it!

datapythonista commented:
Thank you @CristianoS16 for the contribution, nice work, and sorry for the delay.
